pyomo/util/tests/test_latex_printer.py

```diff
@@ -0,0 +1,349 @@
+        bstr += '    \\begin{aligned} \n'
+        bstr += '        & \\text{minimize} \n'
+        bstr += '        & & x + y \\\\ \n'
+        bstr += '        & \\text{subject to} \n'
+        bstr += '        & & x^{2} + y^{2} \leq 1 \\\\ \n'
+        bstr += '        &&& 0 \leq x \\\\ \n'
+        bstr += '        &&&  \left( x + y \\right) \sum_{i \\in I} v_{i} + u_{i,j}^{2} \leq 0 , \quad j \\in I \\\\ \n'
+        bstr += '        &&& \sum_{k \\in K} p_{k} = 1 \n'
+        bstr += '    \end{aligned} \n'
+        bstr += '    \label{basicFormulation} \n'
+        bstr += '\end{equation} \n'
+        self.assertEqual(pstr, bstr)
+
+        pstr = latex_printer(m, None, True)
+        bstr = ''
+        bstr += '\\begin{align} \n'
+        bstr += '    & \\text{minimize} \n'
+        bstr += '    & & x + y \label{obj:basicFormulation_objective_1} \\\\ \n'
+        bstr += '    & \\text{subject to} \n'
+        bstr += '    & & x^{2} + y^{2} \leq 1 \label{con:basicFormulation_constraint_1} \\\\ \n'
+        bstr += '    &&& 0 \leq x \label{con:basicFormulation_constraint_2} \\\\ \n'
+        bstr += '    &&&  \left( x + y \\right) \sum_{i \\in I} v_{i} + u_{i,j}^{2} \leq 0 , \quad j \\in I \label{con:basicFormulation_constraint_7} \\\\ \n'
+        bstr += '    &&& \sum_{k \\in K} p_{k} = 1 \label{con:basicFormulation_constraint_8} \n'
+        bstr += '\end{align} \n'
+        self.assertEqual(pstr, bstr)
+
+        pstr = latex_printer(m, None, False, True)
+        bstr = ''
+        bstr += '\\begin{equation} \n'
+        bstr += '    \\begin{aligned} \n'
+        bstr += '        & \\text{minimize} \n'
+        bstr += '        & & x + y \\\\ \n'
+        bstr += '        & \\text{subject to} \n'
+        bstr += '        & & x^{2} + y^{2} \leq 1 \\\\ \n'
+        bstr += '        &&& 0 \leq x \\\\ \n'
+        bstr += '        &&&  \left( x + y \\right) \sum_{i = 1}^{5} v_{i} + u_{i,j}^{2} \leq 0 , \quad j \\in I \\\\ \n'
+        bstr += '        &&& \sum_{k \\in K} p_{k} = 1 \n'
+        bstr += '    \end{aligned} \n'
+        bstr += '    \label{basicFormulation} \n'
+        bstr += '\end{equation} \n'
+        self.assertEqual(pstr, bstr)
+
+        pstr = latex_printer(m, None, True, True)
+        bstr = ''
+        bstr += '\\begin{align} \n'
+        bstr += '    & \\text{minimize} \n'
+        bstr += '    & & x + y \label{obj:basicFormulation_objective_1} \\\\ \n'
+        bstr += '    & \\text{subject to} \n'
+        bstr += '    & & x^{2} + y^{2} \leq 1 \label{con:basicFormulation_constraint_1} \\\\ \n'
+        bstr += '    &&& 0 \leq x \label{con:basicFormulation_constraint_2} \\\\ \n'
+        bstr += '    &&&  \left( x + y \\right) \sum_{i = 1}^{5} v_{i} + u_{i,j}^{2} \leq 0 , \quad j \\in I \label{con:basicFormulation_constraint_7} \\\\ \n'
+        bstr += '    &&& \sum_{k \in K} p_{k} = 1 \label{con:basicFormulation_constraint_8} \n'
+        bstr += '\end{align} \n'
+        self.assertEqual(pstr, bstr)
+
```

Comment:
Learn some more Pyomo funness! You can write a test to ensure that it outputs to a file using the `TempfileManager`.

Reply:
Fixed